update the kernel config documentation to use `nix-shell`

fricklerhandwerk opened the pull request:
this also slightly improves the instructions after working through them
to check that they do what one would expect.

co-authored with @StefanSchroeder and @proofconstruction

bjornfor commented on doc/builders/packages/linux.section.md:
````diff
@@ -18,23 +18,38 @@ How to add a new (major) version of the Linux kernel to Nixpkgs:
+
+          ```ShellSession
+          $ nix-shell -p ncurses
+          $ export NIX_CFLAGS_LINK=-lncurses
````
Slightly off-topic: I think `nix-shell -p ncurses -p pkg-config` allows dropping `NIX_CFLAGS_LINK`.

DanielSidhion commented on doc/builders/packages/linux.section.md:
```diff
@@ -18,23 +18,38 @@ How to add a new (major) version of the Linux kernel to Nixpkgs:
+    3. Then for each supported platform (`i686`, `x86_64`, `uml`) do the following:
 
-    5.  Copy `.config` over the new config file (e.g., `config-2.6.22-i686-smp`).
+       1. Make a copy from the old config (e.g., `config-2.6.21-i686-smp`) to the new one (e.g., `config-2.6.22-i686-smp`).
+
+       2. Copy the config file for this platform (e.g., `config-2.6.22-i686-smp`) to `.config` in the unpacked kernel source tree.
```
I just decided to go through these instructions because they seemed weird to me - we don't have these config files anymore in `nixpkgs`. I'm afraid these instructions are wildly outdated (as one can guess by the kernel version used in the example), and not relevant anymore. From my experience dealing with the kernel builds in `nixpkgs`, the configs come from [common-config.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/common-config.nix) now. There simply is no `config-*` file that can be used to follow these instructions.

Out of curiosity, I decided to figure out where the instructions come from, and looks like they're at least ~16 years old: https://github.com/NixOS/nixpkgs/blob/d145a966abee342e31c609dddb165d019b1225a7/pkgs/os-specific/linux/kernel/README

I'm not deeply familiar with the process to add new kernels in `nixpkgs`. I only became familiar with some parts of the kernel build process lately, but I'm willing to do some work to figure out up-to-date instructions for this. I'm not sure if this is the right place to do this, but my initial starting point would be to ping folks who seem to be regulars at updating the kernel (can check through commit history) to see if they all follow a particular process.

fricklerhandwerk replied:
I suspected this would be old stuff, but I'd be surprised if one wouldn't be able to make use of the output from the  convenience tool. What do you do about massive configs? Would be impractical to write out each parameter manually.

DanielSidhion replied:
If a previous config file exists, then a `make menuconfig` or similar would likely be helpful. However, since it looks like all of the config is not coming from any files anymore, whoever follows these steps will probably end up having to figure out all the values by themselves again, which is impractical - most of the values are already figured out in `common-config.nix`.

I'm ok with the suggestion to merge this and open another one when we have proper instructions for the current way that we build kernels, since that's likely to take a while anyway (involves contacting other people who may or may not respond quickly).

fricklerhandwerk committed:
doc: update the kernel config documentation to use `nix-shell`

this also slightly improves the instructions after working through them
to check that they do what one would expect.

Co-authored-by: alex <source@proof.construction>
